pgmanage/app/static/pgmanage_frontend/src/components/FileManager.vue

```diff
@@ -399,8 +402,39 @@ export default {
           showToast("error", error.response.data.data);
         });
     },
-    //TODO: implement upload functionality
-    onUpload() {},
+    async handleFileUpload(event) {
+      const file = event.target.files[0];
+
+      if (file.size > maxFileSizeInKB) {
```

Comment:
this should be a separate, externally configurable variable. Define MAX_UPLOAD_SIZE in custom_settings.py and use it here.